pkg/admin/server/apis.go

```diff
@@ -81,48 +82,57 @@ func configDump(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintf(w, "only support one parameter")
 		return
 	}
-	var info interface{}
+	handle := func(info interface{}) {
+		if info == nil {
+			log.DefaultLogger.Alertf(types.ErrorKeyAdmin, "api: %s, parameters:%v", "config dump", r.Form)
+			w.WriteHeader(500)
+			msg := fmt.Sprintf(errMsgFmt, "internal error")
+			fmt.Fprint(w, msg)
+		} else {
+			buf, _ := json.MarshalIndent(info, "", " ")
+			log.DefaultLogger.Infof("[admin api] [config dump] config dump, parameters:%v", r.Form)
+			w.WriteHeader(200)
+			w.Write(buf)
+		}
+	}
 	for key, param := range r.Form {
 		switch key {
 		case "mosnconfig":
-			info = store.GetMOSNConfig(store.CfgTypeMOSN)
+			configmanager.HandleMOSNConfig(configmanager.CfgTypeMOSN, handle)
 		case "allrouters":
-			info = store.GetMOSNConfig(store.CfgTypeRouter)
+			configmanager.HandleMOSNConfig(configmanager.CfgTypeRouter, handle)
 		case "allclusters":
-			info = store.GetMOSNConfig(store.CfgTypeCluster)
+			configmanager.HandleMOSNConfig(configmanager.CfgTypeCluster, handle)
 		case "alllisteners":
-			info = store.GetMOSNConfig(store.CfgTypeListener)
+			configmanager.HandleMOSNConfig(configmanager.CfgTypeListener, handle)
 		case "router":
-			v := store.GetMOSNConfig(store.CfgTypeRouter)
-			routerInfo, ok := v.(map[string]v2.RouterConfiguration)
-			if ok && len(param) > 0 {
-				info = routerInfo[param[0]]
-			}
+			configmanager.HandleMOSNConfig(configmanager.CfgTypeRouter, func(v interface{}) {
+				routerInfo, ok := v.(map[string]v2.RouterConfiguration)
+				if ok && len(param) > 0 {
+					handle(routerInfo[param[0]])
```

Comment:
The function literal capture the loop variable, and always use the last param of  r.Form?